core/capabilities/ccip/common/addresscodec.go

```diff
@@ -0,0 +1,77 @@
+package common
+
+import (
+	"fmt"
+
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+type ChainSpecificAddressCodec interface {
+	AddressBytesToString([]byte) (string, error)
+	AddressStringToBytes(string) ([]byte, error)
+}
+
+type AddressCodec struct {
+	EVMAddressCodec    ChainSpecificAddressCodec
+	SolanaAddressCodec ChainSpecificAddressCodec
+}
+
+// AddressCodecParams is a struct that holds the parameters for creating a AddressCodec
+type AddressCodecParams struct {
+	evmAddressCodec    ChainSpecificAddressCodec
+	solanaAddressCodec ChainSpecificAddressCodec
+}
+
+// NewAddressCodecParams is a constructor for AddressCodecParams
+func NewAddressCodecParams(evmAddressCodec ChainSpecificAddressCodec, solanaAddressCodec ChainSpecificAddressCodec) AddressCodecParams {
+	return AddressCodecParams{
+		evmAddressCodec:    evmAddressCodec,
+		solanaAddressCodec: solanaAddressCodec,
+	}
+}
+
+// NewAddressCodec is a constructor for AddressCodec
+func NewAddressCodec(params AddressCodecParams) AddressCodec {
```

Comment:
I don't think the AddressCodecParams struct is needed.

Just do this:

	return AddressCodec{
		EVMAddressCodec:    ccipevm.AddressCodec{},
		SolanaAddressCodec: ccipsolana.AddressCodec{},
	}

Reply:
this will introduce import cycle